### ISSUE TITLE
Remove Prerender constraint from UI type alias

### ISSUE THREAD

**Issue by mightybyte** [tomsmalley/semantic-reflex#42]
A lot of the semantic-reflex widgets don't need the `Prerender` constraint, but they use the `UI` type alias so they get it.  The result is that the user can't use these widgets in many places completely unnecessarily.

**Comment by tomsmalley** [comment]
The `Prerender` constraint is necessary due to transitions needing `MonadIO` for delaying events.

What are the situations that prevent you from making use of the prerender constraint?

**Comment by tomsmalley** [comment]
It would be nice to be able to drop that constraint where transitions aren't being used, though.

**Comment by mightybyte** [comment]
Yes.  I don't object to the constraint being there where needed.  I just think it shouldn't be there across the board.
